### ISSUE TITLE
[BUG] incidence matrix not working ?

### ISSUE THREAD

**Issue by gdalle** [JuliaGraphs/LightGraphs.jl#1489]
**Description of bug**

Hi there,
I was trying to solve a min cost flow problem using the incidence matrix of a `SimpleDiGraph`, and I got the impression that the function `incidence_matrix` is incorrect.

**How to reproduce**

```julia
julia> using LightGraphs

julia> function my_incidence_matrix(G)
           M = zeros(Int, nv(G), ne(G))
           for (k, e) in enumerate(edges(G))
               M[src(e), k] = -1
               M[dst(e), k] = +1
           end
           return M
       end
my_incidence_matrix (generic function with 1 method)

julia> G = complete_digraph(3)
{3, 6} directed simple Int64 graph

julia> M1 = Matrix(incidence_matrix(G))
3×6 Array{Int64,2}:
 -1  -1   1  -1   1  -1
  1   0  -1   0   0   1
  0   1   0   1  -1   0

julia> M2 = my_incidence_matrix(G)
3×6 Array{Int64,2}:
 -1  -1   1   0   1   0
  1   0  -1  -1   0   1
  0   1   0   1  -1  -1

julia> M1 .== M2
3×6 BitArray{2}:
 1  1  1  0  1  0
 1  1  1  0  1  1
 1  1  1  1  1  0
```

**Expected behavior**

Both "incidence matrices" (the one I computed and the one computed by `LightGraphs`) should be equal.

**Actual behavior**

They are different, in particular the one computed by `LightGraphs` has no zeros on the first row, even though vertex 1 is clearly not incident to all edges.

**Code demonstrating bug**

See above

**Version information**
 
`Julia Version 1.5.2
Commit 539f3ce943 (2020-09-23 23:17 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-9.0.1 (ORCJIT, skylake)
Environment:
  JULIA_NUM_THREADS = 6`
 
`Status `~/.julia/environments/v1.5/Project.toml`
  [093fc24a] LightGraphs v1.3.3`

**Additional context**

I think the bug may be due to lines 170-174 of `spectral.jl`, I don't understand their purpose and commenting them solves it.

**Comment by sbromberger** [comment]
cc @jpfairbanks - would you mind taking a look at this?

**Comment by gdalle** [comment]
Btw, I wrote a faster version of `my_incidence_matrix`, not sure it helps but anyway

```julia
function my_incidence_matrix(G)
    I = vcat([src(e) for e in edges(G)], [dst(e) for e in edges(G)])
    J = vcat(collect(1:ne(G)), collect(1:ne(G)))
    V = vcat(fill(-1, ne(G)), fill(1, ne(G)))
    return sparse(I, J, V)
end
```

**Comment by sbromberger** [comment]
170 - 174 were added to fix #1251 .

**Comment by gdalle** [comment]
Hi,
Does anyone know if this can be fixed easily? If it is really a bug, it is quite annoying for flow problems.
Thanks :)

**Comment by jpfairbanks** [comment]
The snippet above looks right, if the existing code is incorrect, we should replace it with that snippet and fix any performance problems latter.

**Comment by gdalle** [comment]
I have never done a PR, should I try to submit one myself?

**Comment by jpfairbanks** [comment]
yeah, you just:

1. make a fork 
2. clone the fork
3. `git checkout -b gdalle/incidence_matrix`
4. make the changes
5. `git commit -am "fix #1489"`
6. `git push`
7. go to the LG github page and it will invite you to open the PR.

**Comment by gdalle** [comment]
Hey @sbromberger, is it just me or is this bug still bugging, at least on the stable branch of LightGraphs? I thought you had solved it with your PR but I still get the same wrong results when using `incidence_matrix` :thinking:
